src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/image.py

```
@@ -0,0 +1,50 @@
+from azure.cli.core.commands import LongRunningOperation
+from azure.mgmt.containerregistry.v2018_02_01_preview.models import (
+    ImportImageParameters,
+    ImportSource
+)
+from ._utils import (
+    validate_managed_registry,
+    get_resource_id_by_registry_name
+)
+
+from ._client_factory import cf_acr_registries
+
+IMPORT_NOT_SUPPORTED = "Image imports are only supported for managed registries."
+
+def acr_image_import(cmd,
+                     registry_name,
+                     image=None,
+                     resource_id=None,
+                     source_image=None,
+                     target_tags=None,
+                     resource_group_name=None,
+                     untagged_target_repositories=None,
+                     mode="NoForce"):
+    registry, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, IMPORT_NOT_SUPPORTED)
+
+    client_registries = cf_acr_registries(cmd.cli_ctx)
+
+    if image is not None:
+        source_registry = image[:image.index(".")]
```

Would expect more error handling here such as there is no `.`

Also the flow in this block doesn't look right. What happens if `image` is empty, `resource_id` is `None`, and `source_image` is `None`. Wouldn't line 34 throw?

---

If the image, resource_id and source_image are both None, then CLI thinks that customer does not provide a source image to be imported. Therefore, we need to throw an error here. How does it sound? @sajayantony

---

Source image is a required arg.